### ISSUE TITLE
wish a classify.py

### ISSUE THREAD

**Issue by ShengWang11** [sujitpal/fttl-with-keras#2]
@sujitpal Hello,sir.
I am trying to use the trained model to creat  classify.py which can  classify the input picture and display the probability. But  the problem showed in problem.png couldn't be solved after my efforts. Related files has been offered below.
[problem](https://cloud.githubusercontent.com/assets/24971999/26451979/cfbf70c2-4190-11e7-83e0-37b16c7fdb5f.PNG)  
Wish you can take a few minutes to make the classify.py work.  I'll be great appreciated with that.
My email 1178567190@qq.com for timely communication. Thanks for your sharing again.

**Comment by ShengWang11** [comment]
@sujitpal Sorry sir,the classify.py uploading failed. So I copy the code to this comment. 
If necessary, I'd like to send you related files by email.Would you mind leaving your email address or some ways I can send you the files.
## classify.py
#USAGE     python classify.py --image images/2/192_right.jpeg
import sys
import argparse
import numpy as np
from PIL import Image
import requests
from io import BytesIO
import matplotlib.pyplot as plt
import os
import cv2

from keras.utils import np_utils
from keras.models import load_model
from keras.preprocessing import image
from keras.applications.vgg16 import preprocess_input, decode_predictions

DATA_DIR = "../files"
MODEL_DIR = os.path.join(DATA_DIR, "models")

model = load_model(os.path.join(MODEL_DIR, "tl-dl1-model-final.h5"))
#model = ResNet50(weights='imagenet')
target_size = (224, 224)

def predict(model, img, target_size, top_n=3):
  """Run model prediction on image
  Args:
    model: keras model
    img: PIL format image
    target_size: (w,h) tuple
    top_n: # of top predictions to return
  Returns:
    list of predicted labels and their probabilities
  """
  if img.size != target_size:
    img = img.resize(target_size)

  x = image.img_to_array(img)
  x = np.expand_dims(x, axis=0)
  x = preprocess_input(x)
  x = cv2.resize(x,(25088,224))
  preds = model.predict(x)
  return np_utils.categorical_probas_to_classes(preds) #return decode_predictions(preds, top=top_n)[0]

def plot_preds(image, preds):
  """Displays image and the top-n predicted probabilities in a bar graph
  Args:
    image: PIL image
    preds: list of predicted labels and their probabilities
  """
  plt.imshow(image)
  plt.axis('off')

  plt.figure()
  order = list(reversed(range(len(preds))))
  bar_preds = [pr[2] for pr in preds]    //////////////////////////////////////// the problem arose here
  labels = (pr[1] for pr in preds)            
  plt.barh(order, bar_preds, alpha=0.5)
  plt.yticks(order, labels)
  plt.xlabel('Probability')
  plt.xlim(0,1.01)
  plt.tight_layout()
  plt.show()

if __name__=="__main__":
  a = argparse.ArgumentParser()
  a.add_argument("--image", help="path to image")
  args = a.parse_args()

  if args.image is None and args.image_url is None:
    a.print_help()
    sys.exit(1)

  if args.image is not None:
    img = Image.open(args.image)
    preds = predict(model, img, target_size)
    plot_preds(img, preds)

**Comment by sujitpal** [comment]
The problem seems to be legit. The predict() function returns an array of predictions. It looks the assumption in the code is that pred is a triple of (_, label, pred) (or your index offset in the assignment to bar_preds and labels are off by 1).
